newrelic/api/opentelemetry.py

```diff
@@ -0,0 +1,486 @@
+            _logger.debug("otel span: %s\nnewrelic trace: %s", self.otel_parent, current_nr_trace)
+            raise ValueError("Unexpected span parent scenario encountered")
+
+        if nr_trace_type == FunctionTrace:
+            trace_kwargs = {"name": self.name, "params": self.attributes, "parent": self.nr_parent}
+            self.nr_trace = nr_trace_type(**trace_kwargs)
+        elif nr_trace_type == DatastoreTrace:
+            trace_kwargs = {
+                "product": self.instrumenting_module,
+                "target": None,
+                "operation": self.name,
+                "parent": self.nr_parent,
+            }
+            self.nr_trace = nr_trace_type(**trace_kwargs)
+        elif nr_trace_type == ExternalTrace:
+            trace_kwargs = {
+                "library": self.name or self.instrumenting_module,
+                "url": self.attributes.get("http.url"),
+                "method": self.attributes.get("http.method"),
+                "parent": self.nr_parent,
+            }
+            self.nr_trace = nr_trace_type(**trace_kwargs)
+        elif nr_trace_type == MessageTrace:
+            trace_kwargs = {
+                "library": self.instrumenting_module,
+                "operation": "Produce",
+                "destination_type": "Topic",
+                "destination_name": self.name,
+                "params": self.attributes,
+                "parent": self.nr_parent,
+                "terminal": False,
+            }
+            self.nr_trace = nr_trace_type(**trace_kwargs)
+        else:
+            # TODO: Still need to implement GraphQLOperationTrace and GraphQLResolverTrace
+            trace_kwargs = {"name": self.name, "params": self.attributes, "parent": self.nr_parent}
+            self.nr_trace = nr_trace_type(**trace_kwargs)
+
+        self.nr_trace.__enter__()
+
+    def _is_sampled(self):
+        # Uses NR to determine if the trace is sampled
+
+        # transaction.sampled can be None, True, False.
+        # If None, this has not been computed by NR which
+        # can also mean the following:
+        # 1. There was not a context passed in that explicitly has sampling disabled.
+        #   This flag would be found in the traceparent or traceparent and tracespan headers.
+        # 2. Transaction was not created where DT headers are accepted during __init__
+        # Therefore, we will treat a value of `None` as `True` for now.
+
+        if self.otel_parent:
+            return bool(self.otel_parent.trace_flags)
+        else:
+            return bool(self.nr_transaction and (self.nr_transaction.sampled or (self.nr_transaction._sampled is None)))
+
+    def _is_remote(self):
+        # Remote span denotes if propagated from a remote parent
+        if self.otel_parent and self.otel_parent.is_remote:
+            return True
+        return False
+
+    def get_span_context(self):
+        if not (hasattr(self, "nr_trace") and self.nr_trace):
```

Comment:
This is hard to follow-maybe do this instead?
```suggestion
        if not getattr(self, "nr_trace", False):
```